tests/ci/run_valgrind_tests.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash -xu
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+################################################################################
+# set -e is disabled in this script because we need the script to continue
+# even if |run_valgrind| returns an error.
+#
+# Assumes execution from root of AWS-LC folder.
+################################################################################
+
+source tests/ci/common_posix_setup.sh
+
+SCRIPT_EXIT_STATUS=0
+
+VALGRIND_COMMAND=valgrind
+VALGRIND_ERROR_CODE=99
+
+# crypto_test setup
+# Suppression file produced by running:
+# valgrind --gen-suppressions=yes ./test_build_dir/crypto/crypto_test
+CRYPTO_TEST_EXE=test_build_dir/crypto/crypto_test
+VALGRIND_SUPPRESSIONS_FILE_CRYPTO_TEST=tests/ci/valgrind_suppressions_crypto_test.supp
+
+################################################################################
+# Input:
+#   1: Executable to be executed under Valgrind
+#   2: Valgrind suppression file for $1
+# Output:
+#   0                       if Valgrind exists without any errors
+#   ${VALGRIND_ERROR_CODE}  if Valgrind exists with errors
+################################################################################
+run_valgrind () {
+
+    TEST_EXE=$1
+    VALGRIND_SUPPRESSIONS_FILE=$2
+    VALGRIND_ARGS=("--error-exitcode=${VALGRIND_ERROR_CODE}" "--track-origins=yes" "--leak-check=full" "--trace-children=yes" "--suppressions=${VALGRIND_SUPPRESSIONS_FILE}")
```

Comment:
* Similar to sanitizer, move these compiler flavors to `CMakeLists.txt` and then creates a flag to control enable/disable?
* This seems duplicate. Is Valgrind not integrated?
```go
# runner.go
	useValgrind              = flag.Bool("valgrind", false, "If true, run code under valgrind")

# all_tests.go
func valgrindOf(dbAttach bool, path string, args ...string) *exec.Cmd {
	valgrindArgs := []string{"--error-exitcode=99", "--track-origins=yes", "--leak-check=full", "--quiet"}
```

Reply:
> You can't specify a suppression file

Why can't?  Either add the path in existing config, or create new test target.

> You can't specify a test json file (by default it takes all targets in all_tests.json.

Add env variable to config diff test json file? Besides, CryptoAlg-597 mentions some key words **sanitizer**, **all tests** and **comparison**. Current sanitizer executes the `all_tests.go`.

> which would take many hours.

Where is this tested? Or this is Valgrind specific? Current sanitizer is executed in BUILD_GENERAL1_LARGE, which has 8 vCPU. The sanitizer takes 40 min to finish build (4 times) and `all_tests.go` (3 times).

Reply:
I could have extended the Go script. I didn't because it was quicker for me to do it bash and there is no mandate to double down on the Go script.

If this gets implemented it means there are two places to configure tests suites.